xcm/xcm-builder/src/universal_exports.rs

```diff
@@ -0,0 +1,368 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Traits and utilities to help with origin mutation and bridging.
+
+use frame_support::{ensure, traits::Get};
+use parity_scale_codec::{Decode, Encode};
+use sp_std::{convert::TryInto, marker::PhantomData, prelude::*};
+use xcm::prelude::*;
+use xcm_executor::traits::{validate_export, ExportXcm};
+use SendError::*;
+
+fn ensure_is_remote(
+	universal_local: impl Into<InteriorMultiLocation>,
+	dest: impl Into<MultiLocation>,
+) -> Result<(NetworkId, InteriorMultiLocation), MultiLocation> {
+	let dest = dest.into();
+	let universal_local = universal_local.into();
+	let local_net = match universal_local.global_consensus() {
+		Ok(x) => x,
+		Err(_) => return Err(dest),
+	};
+	let universal_destination: InteriorMultiLocation = universal_local
+		.into_location()
+		.appended_with(dest.clone())
+		.map_err(|x| x.1)?
+		.try_into()?;
+	let (remote_dest, remote_net) = match universal_destination.split_first() {
+		(d, Some(GlobalConsensus(n))) if n != local_net => (d, n),
+		_ => return Err(dest),
+	};
+	Ok((remote_net, remote_dest))
+}
+
+/// Implementation of `SendXcm` which uses the given `ExportXcm` implementation in order to forward
+/// the message over a bridge.
+///
+/// No effort is made to charge for any bridge fees, so this can only be used when it is known
+/// that the message sending cannot be abused in any way.
+///
+/// This is only useful when the local chain has bridging capabilities.
+pub struct LocalUnpaidExporter<Exporter, UniversalLocation>(
```

Comment:
Nit: What about align naming here with UnpaidRemoteExporter / SovereignPaidRemoteExporter?
```suggestion
pub struct UnpaidLocalExporter<Exporter, UniversalLocation>(
```